extensions/resource-deployment/src/ui/validation/validations.ts

```diff
@@ -0,0 +1,221 @@
+		super(validation, valueGetter, variableValueGetter);
+		throwUnless(validation.target !== undefined);
+		this._target = validation.target;
+	}
+
+	abstract isComparisonSuccessful(): Promise<boolean>;
+	validate: Validator = async () => {
+		const isValid = await this.isComparisonSuccessful();
+		return Promise.resolve({
+			valid: isValid,
+			message:  isValid ? undefined: this.description
+		});
+	};
+}
+
+export class LessThanOrEqualsValidation extends Comparison {
+	async isComparisonSuccessful() {
+		return (await this.getValue())! <= ((await this.getVariableValue(this.target))!);
+	}
+}
+
+export class GreaterThanOrEqualsValidation extends Comparison {
+	async isComparisonSuccessful() {
+		return (await this.getValue())! >= ((await this.getVariableValue(this.target))!);
+	}
+}
+
+
+/**
+ *	removes validation message corresponding to this validator from the @see dialogMessage string.
+ *
+ * @param dialogMessage
+ * @param message
+*/
+export function removeValidationMessage(dialogMessage: azdata.window.DialogMessage, message: string): azdata.window.DialogMessage {
+	if (dialogMessage === undefined) {
+		return dialogMessage;
+	}
+	if (dialogMessage.description) {
+		return {
+			text: dialogMessage.text,
+			level: dialogMessage.level,
+			description: getStrippedMessage(dialogMessage.description, message)
+		};
+	} else {
+		return {
+			text: getStrippedMessage(dialogMessage.text, message),
+			level: dialogMessage.level,
+			description: dialogMessage.description
+		};
+	}
+}
+
+function getStrippedMessage(originalMessage: string, message: string) {
+	if (originalMessage.includes(message)) {
+		const messageWithLineBreak = message + EOL;
+		const searchText = originalMessage.includes(messageWithLineBreak) ? messageWithLineBreak : message;
+		const newMessage = originalMessage.replace(searchText, '');
+		return newMessage;
+	} else {
+		return originalMessage;
+	}
+}
+
+export function createValidation(validation: ValidationInfo, valueGetter: ValueGetter, variableValueGetter?: VariableValueGetter): Validation {
+	switch (validation.type) {
+		case ValidationType.Regex: return new RegexValidation(<RegexValidationInfo>validation, valueGetter);
+		case ValidationType.IsInteger: return new IntegerValidation(<IntegerValidationInfo>validation, valueGetter);
+		case ValidationType.LessThanOrEqualsTo: return new LessThanOrEqualsValidation(<ComparisonValidationInfo>validation, valueGetter, variableValueGetter!);
+		case ValidationType.GreaterThanOrEqualsTo: return new GreaterThanOrEqualsValidation(<ComparisonValidationInfo>validation, valueGetter, variableValueGetter!);
+		default: throw new Error(`unknown validation type:${validation.type}`); //dev error
+	}
+}
+
+export async function validateAndUpdateValidationMessages(component: InputComponent, container: azdata.window.Dialog | azdata.window.Wizard, validations: Validation[] = []): Promise<ValidationResult> {
```

Comment:
I think for each component, you should only return the validation failure messages and there should be a centralized location to actually update the container.message.

Reply:
yes this is the centralized location. This is just a convenient helper to do the updates. The validation maessages for a given component are returned by the Validation object and not this call.

Reply:
but you are taking the component as a parameter and update the container.message in this method, right?

Reply:
the component is not really needed for updates to the container.message. I can remove that parameter and deal with that change in the calling code. I think that will make this method clearer as it is currently doing two things. One is updating the container.message and the second is to update the component.validationErrorMessage property

I can move the updating of component.valdationErrorMessage property out to the calling code.

Reply:
I had an offline chat with @alanrenmsft on this. I am going to remove the updating of the dialog message completely from this validation module. Any client code can update it as needed. This simplifies these functions quite a bit and it was Charles' feedback also around showing 2 messages, so this will address that as well. If indeed there is a valid case of 2 messages to be shown, then the client code can do the needful.

Reply:
Checked in simplifications as per discussion here. Now we just update the validationErrorMessage in place within the input field.